apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js

```diff
@@ -0,0 +1,78 @@
+const crypto = require('crypto')
+
+const dayjs = require('dayjs')
+const { map } = require('lodash')
+
+const { allItemsQueryByChunks, itemsQuery } = require('@open-condo/keystone/schema')
+
+const { md5 } = require('@condo/domains/common/utils/crypto')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+
+
+const CHUNK_SIZE = 1000
+const PHONE_REGEXP = /^\+7\d{10}$/
+
+async function getHashedResidentsAndContactsPhones ({ lastSyncDate, toEmail, writePhoneCb }) {
+    const where = {
+        phone_contains_i: '+7',
+        deletedAt: null,
+    }
+    if (lastSyncDate) {
+        where.createdAt_gt = dayjs(lastSyncDate).toISOString()
+    }
+
+    const { count: contactsCount } = await itemsQuery('Contact', {
+        where,
+    }, { meta: true })
+    const { count: residentUsersCount } = await itemsQuery('User', {
+        where: { ...where, type: RESIDENT },
+    }, { meta: true })
+
+    const maxLine = contactsCount + residentUsersCount
+    if (maxLine === 0) {
+        const msg = 'Empty contacts and resident users'
+        throw new Error(msg)
+    }
+
+    const lineWithIdentityHash = crypto.randomInt(maxLine)
+    let lineNum = 0
+
+    const chunkProcessor = async (chunk) => {
+        const phones = map(chunk, 'phone')
+
+        for (const phone of phones) {
+            if (PHONE_REGEXP.test(phone)) {
+                const formattedPhone = phone.slice(1)
+                writePhoneCb(formattedPhone)
+            }
+
+            if (lineNum === lineWithIdentityHash) {
+                // Hash to identify file: when and where was it sent. Example: 2024-05-30-example@example.com
+                const identityHash = md5(`${dayjs().format('YYYY-MM-DD')}-${toEmail}`)
+                writePhoneCb(identityHash)
+            }
```

Comment:
Is this for internal use?

Reply:
Yes, it is for identify file if it was leaked